app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -135,5 +174,60 @@ def remove_status_history(target)
     def get_progress_message(action)
       "#{action == 'done' ? 'Published' : action.titleize}: #{@status_history.fetch(action, {})['action_date'].in_time_zone('UTC').to_s(:default_with_time_zone)} By: #{@status_history.fetch(action, {})['username']}"
     end
+
+    # Fetch e-mails of users who are not the current user from URS to send them
+    # e-mails.
+    def user_from_resource
+      proposal_urs_user = retrieve_urs_users([User.find(get_resource.user_id).urs_uid])[0]
+      { name: "#{proposal_urs_user['first_name']} #{proposal_urs_user['last_name']}", email: proposal_urs_user['email_address'] } unless proposal_urs_user.blank?
+    end
+
+    # Used as a before action to manipulate which look of the role based
+    # two-look pages is used.
+    def approver_status
+      @non_nasa_approver = is_non_nasa_draft_approver?(user: current_user, token: token)
+    end
+
+    def get_available_actions_text
+      if get_resource.in_work?
+        'Make additional changes or submit this proposal for approval.'
+      elsif @non_nasa_approver
+        if get_resource.approved?
+          'Please visit the Metadata Management Tool for NASA users to finish publishing this metadata.'
+        elsif get_resource.done?
+          'No actions are possible.'
+        end
+      elsif get_resource.submitted? || get_resource.rejected?
+        'You may rescind this proposal to make additional changes.'
+      else
+        'No actions are possible.'
+      end
+    end
+
+    # Pass in the parameters of the where clause to restrict which proposals are
+    # displayed.  Passing nil for both is valid and returns all proposals.
+    # TODO: Investigate if this can be used to provide sorting to all drafts.
```

Comment:
I'll take the action to write up the standalone ticket if we agree.  If it is really deemed necessary, I can go back and do it in this ticket.